File: libecole/tests/src/test-configuring.cpp
#include <memory>
#include <stdexcept>

#include <catch2/catch.hpp>

#include "ecole/configuring.hpp"
#include "ecole/observation.hpp"

#include "conftest.hpp"

using namespace ecole;

TEST_CASE("Model creation") {
	auto env = configuring::Env<int, obs::BasicObs<>>(
		std::make_unique<obs::BasicObsFunction<>>(),
		std::make_unique<configuring::Configure<int>>("conflict/lpiterations"));

	for (auto i = 0L; i < 2; ++i) {
		env.reset(problem_file);
		env.step(0);
	}
}

File: python/tests/environment/test_configuring.py
import pytest

import ecole.configuring


def test_Configure(model):
    conf = ecole.configuring.Configure("conflict/lpiterations")
    conf.set(model, 0)


def test_Configure_char(model):
    conf = ecole.configuring.Configure("branching/scorefunc")
    conf.set(model, "s")


def test_Configure_error(model):
    conf = ecole.configuring.Configure("conflict/lpiterations")
    # FIXME test with scip.Exception
    with pytest.raises(Exception):
        conf.set(model, [1, 2])


@pytest.mark.slow
def test_configuring_environment(model):
    env = ecole.configuring.Environment.make_dummy("conflict/lpiterations")
    for _ in range(2):
        count = 0
        obs, done = env.reset(model.clone())
        while not done:
            obs, reward, done, info = env.step(0)
            count += 1
        assert count == 1